backend/compact-connect/lambdas/provider-data-v1/tests/function/test_data_model/test_client.py

```diff
@@ -0,0 +1,218 @@
+        """
+        Because of the risk of exposing sensitive data to the public if we manage to get corrupted
+        data into our database, we'll specifically validate data coming _out_ of the database
+        and throw an error if it doesn't look as expected.
+        """
+        from data_model.client import DataClient
+        from exceptions import CCInternalException
+
+        provider_id = self._load_provider_data()
+
+        with open('tests/resources/dynamo/license.json', 'r') as f:
+            license_record = json.load(f)
+
+        self._table.put_item(
+            Item={
+                # Oh, no! We've somehow put somebody's SSN in the wrong place!
+                'something_unexpected': '123-12-1234',
+                **license_record
+            }
+        )
+
+        client = DataClient(self.config)
+
+        # This record should not be allowed out via API
+        with self.assertRaises(CCInternalException):
+            client.get_provider(  # pylint: disable=missing-kwoa,unexpected-keyword-arg
+                compact='aslp',
+                provider_id=provider_id
+            )
+
+    def test_get_licenses_sorted_by_family_name(self):
+        from data_model.client import DataClient
+
+        self._generate_providers(home='oh', privilege='ne', start_serial=9999)
+        self._generate_providers(home='ne', privilege='oh', start_serial=9989)
+        self._generate_providers(home='ne', privilege='co', start_serial=9979)
+        client = DataClient(self.config)
+
+        # We expect to see 20 providers: 10 have privileges in oh, 10 have licenses in oh
+        resp = client.get_licenses_sorted_by_family_name(  # pylint: disable=unexpected-keyword-arg,missing-kwoa
+            compact='aslp',
+            jurisdiction='oh',
+            pagination={
+                'pageSize': 10
+            }
+        )
+        first_provider_ids = {item['providerId'] for item in resp['items']}
+        first_provider_items = resp['items']
+        self.assertEqual(10, len(resp['items']))
+        self.assertIsInstance(resp['pagination']['lastKey'], str)
+
+        last_key = resp['pagination']['lastKey']
+        resp = client.get_licenses_sorted_by_family_name(  # pylint: disable=unexpected-keyword-arg,missing-kwoa
+            compact='aslp',
+            jurisdiction='oh',
+            pagination={'lastKey': last_key, 'pageSize': 100}
+        )
+        self.assertEqual(10, len(resp['items']))
+        self.assertIsNone(resp['pagination']['lastKey'])
+
+        second_provider_ids = {item['providerId'] for item in resp['items']}
+        # Verify that there are no repeat items between the two calls
+        self.assertFalse(first_provider_ids & second_provider_ids)
+
+        all_items = [*first_provider_items, *resp['items']]
+        first_item = all_items[0]
+        # Verify sorting by family name
+        for item in all_items:
+            self.assertLessEqual(first_item['familyName'], item['familyName'])
```

Comment:
That's a good way to do it. In my head, it just seemed more explicit what order I was testing for. I can switch it over.

Reply:
Sounds good. I think it makes sense to switch, since a list like `["A", "C", "D", "B", "E"]` would seemingly pass this test, even though it isn't sorted.

If we switch, mind doing it for all the tests in other files involving sorting, too? I think there are a couple.

Reply:
That list would fail this test - it is a strict test of sorting, just in more statements than your provided method uses. I will update all of them.

Reply:
Sounds good! (I admit I must be missing something re: the current approach, but alas-onward and upward.)